qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/FieldCapsIT.java

```diff
@@ -272,18 +272,20 @@ public void testAllIndicesWithIndexFilter() throws Exception {
     @SuppressWarnings("unchecked")
     // Returns a client connected to one of the upgraded nodes.
     private RestClient getUpgradedNodeClient() throws IOException {
+        // TODO[lor]: review this after we moved NodeInfo to qualified ids (Build.current().version())
+        var currentVersion = Build.current().version().replace("-SNAPSHOT", "");
```

Comment:
I'll answer here for all the "unqualified" related comments;
Yes, the whole replace `("-SNAPSHOT", "")`  is related to either comparing things that still use `Version.CURRENT.toString()` (e.g. NodeInfo) to `Build.current().version()` (which includes the suffix), or anticipanting a change - moving NodeInfo from Version.CURRENT to `Build.current().version()` will mean that afterwards we need to remove the suffix, and/or go again over all these replace() and remove them once we don't have a mix anymore.

Indeed this all looks like a bug: we have both qualifiedVersion() and version() returning the same suffixed version.

Reply:
The good thing of having the string with suffix returned by `version()` was that it uncovered all the places where it was parsed to a Version :) 
But now that we have them, it would save a lot of ugly `replace()`s to have `version()` returning the unqualified string.

Reply:
After https://github.com/elastic/elasticsearch/pull/100868 the replace is not needed anymore, removed.